pkg/virt-handler/vm.go

```diff
@@ -761,6 +767,27 @@ func (d *VirtualMachineController) generateEventsForVolumeStatusChange(vmi *v1.V
 	}
 }
 
+func (d *VirtualMachineController) updateDiskSize(vmi *v1.VirtualMachineInstance, volumeStatus v1.VolumeStatus, specVolumeMap map[string]v1.Volume) v1.VolumeStatus {
+	var name string
+	source := specVolumeMap[volumeStatus.Name]
+	if source.PersistentVolumeClaim != nil {
+		name = source.PersistentVolumeClaim.ClaimName
+	} else if source.DataVolume != nil {
+		name = source.DataVolume.Name
+	} else {
+		return volumeStatus
+	}
+	pvc, err := d.clientset.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), name, metav1.GetOptions{})
+	if err != nil {
+		return volumeStatus
+	}
+	expectedDiskSize, ok := pvcutils.ExpectedDiskSize(d.cdiConfigInformer.GetStore(), d.scInformer.GetStore(), pvc)
+	if ok && d.clusterConfig.ExpandDisksEnabled() {
+		volumeStatus.Size = &expectedDiskSize
```

Comment:
The value here is ultimately reported on the VMI.Status. If i saw that value, i'd expect that my VMI's disk has been resized, but there's no guarantee that's the case.

This is the expectedSize. The domain may or may not have been resized if there were errors during the resize. I'd expect the status value to be an observation of what occurred on the domain.

Reply:
Changed to ExpectedSize.